via/services/youtube_transcript.py

```diff
@@ -0,0 +1,135 @@
+import re
+from base64 import b64encode
+from dataclasses import dataclass
+from typing import Dict, List
+from xml.etree import ElementTree
+
+from via.services.http import HTTPService
+
+
+@dataclass
+class TranscriptInfo:
+    language_code: str
+    name: str
+    url: str
+    autogenerated: bool
+
+    @property
+    def id(self) -> str:  # pylint:disable=invalid-name
+        """Return a unique ID for this transcript."""
+        name = b64encode(self.name.encode("utf-8")).decode("utf-8")
+
+        return ".".join(
+            part or ""
+            for part in [
+                self.language_code,
+                "a" if self.autogenerated else None,
+                name,
+            ]
+        ).rstrip(".")
+
+
+DEFAULT_TRANSCRIPT_PREFERENCES = (
+    {
+        "language_code": re.compile("en"),
+        "name": re.compile("English"),
+        "autogenerated": False,
+    },
+    {
+        "language_code": re.compile("en-.*"),
+        "name": re.compile(r"English \(.*\)"),
+        "autogenerated": False,
+    },
+    {
+        "language_code": re.compile("en"),
+        "name": re.compile(".*"),
+        "autogenerated": False,
+    },
+    {
+        "language_code": re.compile("en-.*"),
+        "name": re.compile(".*"),
+        "autogenerated": False,
+    },
+    {
+        "language_code": re.compile("en"),
+        "name": re.compile(".*"),
+        "autogenerated": True,
+    },
+    {
+        "language_code": re.compile("en-"),
+        "name": re.compile(".*"),
+        "autogenerated": True,
+    },
+)
```

Comment:
A nod to efficiency here: compiling these regexes at module level so they get compiled just once at application startup.

It's not necessary to anchor these regexes to the starts and ends of the strings with `^` and `$` (or `\A` and `\Z`) because we use `fullmatch()` with them below.